swarm/network/stream/delivery_test.go

```diff
@@ -48,7 +48,7 @@ func TestStreamerRetrieveRequest(t *testing.T) {
 		Retrieval: RetrievalClientOnly,
 		Syncing:   SyncingDisabled,
 	}
-	tester, streamer, _, teardown, err := newStreamerTester(t, regOpts)
```

Comment:
@justelad what @zelig meant here is that if you have access to `testing.T` inside a function, you could call inside `t.Fatal` in case something goes wrong, that you don't care about in the tests.

This basically helps you to get rid of all the `if err != nil` calls in the setup stage of tests, so that you have a more readable test.

Obviously this was not done here, but generally helper functions call `t.Helper()` at the top (so that they report the correct failure line), and then you don't need to return `err` as the test would have failed on the `t.Fatal`

I hope that makes sense.

I suggest we merge this PR after the linter is fixed, but this should be something we improve in the future, to make tests more readable.